src/components/Editor/Runners/HtmlRunner/HtmlRunner.js

```diff
@@ -5,95 +5,127 @@ import { useDispatch, useSelector } from "react-redux";
+    (state) => state.editor.codeRunTriggered
+  );
   const firstPanelIndex = 0;
   const focussedFileIndex = useSelector(
     (state) => state.editor.focussedFileIndices
   )[firstPanelIndex];
-  const openFiles = useSelector((state) => state.editor.openFiles)[firstPanelIndex];
-  const codeRunTriggered = useSelector(
-    (state) => state.editor.codeRunTriggered
-  );
+  const openFiles = useSelector((state) => state.editor.openFiles)[
+    firstPanelIndex
+  ];
   const justLoaded = useSelector((state) => state.editor.justLoaded);
   const isEmbedded = useSelector((state) => state.editor.isEmbedded);
   const autorunEnabled = useSelector((state) => state.editor.autorunEnabled);
   const codeHasBeenRun = useSelector((state) => state.editor.codeHasBeenRun);
 
-  const { t } = useTranslation()
+  const { t } = useTranslation();
   const dispatch = useDispatch();
   const output = useRef();
   const [error, setError] = useState(null);
+  const allowedHrefs = ["#"];
+
+  const focussedComponent = (fileName = "index.html") =>
+    projectCode.filter(
+      (component) => `${component.name}.${component.extension}` === fileName
+    )[0];
+
+  const previewable = (file) => file.endsWith(".html");
+
+  const [previewFile, setPreviewFile] = useState(
+    previewable(openFiles[focussedFileIndex])
+      ? openFiles[focussedFileIndex]
+      : "index.html"
+  );
 
   const showModal = () => {
     dispatch(showErrorModal());
   };
 
   const closeModal = () => setError(null);
 
-  const htmlFiles = projectCode.filter(
-    (component) => component.extension === "html"
-  );
-
-  const fileName = openFiles[focussedFileIndex];
-  const focussedComponent = projectCode.find(
-    (component) => `${component.name}.${component.extension}` === fileName
-  );
-
   const getBlobURL = (code, type) => {
     const blob = new Blob([code], { type });
     return URL.createObjectURL(blob);
   };
 
-  const errorListener = () => {
+  const cssProjectImgs = (projectFile) => {
+    var updatedProjectFile = { ...projectFile };
+    if (projectFile.extension === "css") {
+      projectImages.forEach((image) => {
+        updatedProjectFile.content = updatedProjectFile.content.replace(
+          image.filename,
+          image.url
+        );
+      });
+    }
+    console.log(updatedProjectFile);
+    return updatedProjectFile;
+  };
+
+  const parentTag = (node, tag) =>
+    node.parentNode?.tagName && node.parentNode.tagName.toLowerCase() === tag;
+
+  const eventListener = () => {
     window.addEventListener("message", (event) => {
-      if (typeof event.data === "string" || event.data instanceof String) {
+      if (typeof event.data === "string" || typeof event.data === "string") {
         if (event.data === "ERROR: External link") {
           setError("externalLink");
         }
+      } else if (event.data?.msg && typeof event.data?.msg === "string") {
+        setPreviewFile(`${event.data.payload.linkTo}.html`);
+        dispatch(triggerCodeRun());
```

Comment:
📓 enables us to intercept ALL events and rerun code parsing when a `RELOAD` even is registered